pkg/api/types.go

```diff
@@ -2070,10 +2070,41 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
+// Type and constants for component type.
+type ComponentType string
+
+// These are the valid conditions for the component type.
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
```

Comment:
the constant types are used as unique name prefixes, the same way pod names are.